### PR TITLE
Mod support for all Keypress.

### PR DESCRIPTION
**Pull request by robinvd**
Changes:
- Keypress to allow mods for every type except unknown
- Changes to slate to convert them
- Changes to Vim to ignore mods on keybindings that did not have mods before

Will look into adding things from Vty tomorrow.

### CODE REVIEW

File: rasa-ext-vim/src/Rasa/Ext/Vim.hs [hunk 1]
```diff
@@ -83,10 +83,10 @@ anyMode _ = return ()
 
 -- | Listeners for keypresses when in 'Insert' mode
 insert :: [Keypress] -> BufAction ()
-insert [KEsc] = mode .= Normal
+insert [KEsc _] = mode .= Normal
 
-insert [KBS] = moveRangesByN (-1) >> delete
-insert [KEnter] = insertText "\n" >> moveRangesByC (Coord 1 0) >> startOfLine
+insert [KBS _] = moveRangesByN (-1) >> delete
+insert [KEnter _] = insertText "\n" >> moveRangesByC (Coord 1 0) >> startOfLine
```
Comment (ChrisPenner) [inline rasa-ext-vim/src/Rasa/Ext/Vim.hs]: I'd prefer that we be specific about what we want for these; we probably only want this to act like Enter when there's no mods; same for the others, (probably for insertText too). Can we change these to `KEnter []`  and friends? Thanks!

File: rasa-ext-vim/src/Rasa/Ext/Vim.hs [hunk 2]
```diff
@@ -113,10 +113,10 @@ normal [Keypress 'd' [Ctrl]] = liftAction $ scrollBy 7 -- Half-Page down
...
 normal [Keypress 'u' [Ctrl]] = liftAction $ scrollBy (-7) -- Half-Page up
 
-normal [KLeft] = liftAction focusViewLeft
-normal [KRight] = liftAction focusViewRight
-normal [KUp] = liftAction focusViewAbove
-normal [KDown] = liftAction focusViewBelow
+normal [KLeft _] = liftAction focusViewLeft
+normal [KRight _] = liftAction focusViewRight
+normal [KUp _] = liftAction focusViewAbove
+normal [KDown _] = liftAction focusViewBelow
```
Comment (ChrisPenner) [inline rasa-ext-vim/src/Rasa/Ext/Vim.hs]: Same here; let's be explicit with `KLeft []`

File: rasa-ext-vim/src/Rasa/Ext/Vim.hs (hunk 1 above)
Comment (robinvd) [inline rasa-ext-vim/src/Rasa/Ext/Vim.hs]: Guess it is just a personal preference :). I don't know how close you want to mirror vim with Rasa.Ext.Vim. But vim accepts enter with modifiers.

**Comment by robinvd** [comment]
Changed what you recommended, added keys from Vty and made some examples for them.

**Comment by ChrisPenner** [comment]
Looks great! ✨ 👍 ✨

**Pull request merged**